### PR TITLE
Regional recommender

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

A recommendation service to hint cx to use another region which may cost less after resource creation cmdlets (for now only `New-AzVM`).

The recommendtation service is implenmented as a global component of AzureSession, so it can be called in AzurePSCmdlet, and will work for all SDK-based cmdlets.

We don't have requirements for autorest-based cmdlets yet. If we do, we can add new APIs to the recommendation service. Since it's an abstract class it won't be a binary breaking change.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Authentication/Utilities/Recommendation/RegionalRecommender.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                { "euwest", "UK South" },
+                { "France Central", "North Europe" },
+                { "francecentral", "North Europe" },
+                { "Germany West Central", "North Europe" },
+                { "germanywestcentral", "North Europe" }
+            };
+        }
+
+        /// <inheritdoc/>
+        public bool Process(InvocationInfo invocation, AzurePSQoSEvent qosEvent, out string recommendation)
```
Comment (isra-fel) [inline src/Accounts/Authentication/Utilities/Recommendation/RegionalRecommender.cs:47]: Limit the frequency

**Review by isra-fel — COMMENTED** [review]
Todo: telemetry for azconfig

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Looks good to me with adding change log required

**Comment by isra-fel** [comment]
Common build is ready. Please review @BethanyZhou thanks!

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**